Flexa-Agents/Fallback-State/src/data.fetcher.ts

```diff
@@ -0,0 +1,32 @@
+import { Contract, providers, BigNumber } from "ethers";
+import { flexaInterface } from "./abi";
+import { getEthersProvider } from "forta-agent";
+
+export default class DataFetcher {
+  readonly moduleAddress: string;
+  readonly provider: providers.Provider;
+
+  constructor(moduleAddress: string, provider: providers.Provider) {
+    this.moduleAddress = moduleAddress;
+    this.provider = provider;
+  }
+
+  public async getFallbackSetDate(blockNumber: number, flexaAddress: string): Promise<number> {
+    const flexaContract: Contract = new Contract(flexaAddress, flexaInterface, this.provider);
+    const fallbackSetDate: BigNumber = await flexaContract.fallbackSetDate({ blockTag: blockNumber });
+    return BigNumber.from(fallbackSetDate).toNumber();

```

Comment:
Avoid this conversion. Use `BigNumber` every time you are working with `uint256` because some values will not fit in `number`. Same in the below function.